### PR TITLE
feat: add geofencing zone in map

### PR DESCRIPTION
**Pull request by marius-at-atb**
This is the final PR for adding geofencingZones!
It should work if you test selecting an electric scooter from Voi or Tier (Ryde doesn't have data available atm), and work like this:

<table>
  <tr>
    <td></td>
    <td></td>
     <td></td>
      <td></td>
      <td></td>
       <td></td>
  </tr>
  <tr>
    <td>Allowed</td>
    <td>Slow</td>
    <td>No Parking, isStation=true</td>
    <td>No Parking, isStation=false</td>
    <td>No Entry</td>
    <td>Unspecified</td>
  </tr>
</table>

Depends on https://github.com/AtB-AS/mittatb-app/pull/4582. Only changes from [this commit](https://github.com/AtB-AS/mittatb-app/pull/4583/commits/35e388c122a62bb77fae9d8284e503d8e9776cfe) are new here.

Basically:
- Add GeofencingZones to the map
- Use Snackbar to show explanations
- Handle map onPress events in a better way

Resolves https://github.com/AtB-AS/kundevendt/issues/17614

### CODE REVIEW

File: src/components/map/Map.tsx [hunk 1]
```diff
@@ -86,6 +124,53 @@ export const Map = (props: MapProps) => {
...
+   * Step 3: selected the feature
+   */
+
+  const onFeatureClick = async (feature: Feature) => {
+    if (!isFeaturePoint(feature)) return;
+    const {coordinates: positionClicked} = feature.geometry;
+
+    const featuresAtClick = await getFeaturesAtClick(feature, mapViewRef);
+    if (!featuresAtClick || featuresAtClick.length === 0) return;
+    //console.log('featuresAtClick', JSON.stringify(featuresAtClick));
```
Comment (rosvik) [inline src/components/map/Map.tsx:142]:
```suggestion
```

File: src/components/map/Map.tsx [hunk 2]
```diff
@@ -86,6 +124,53 @@ export const Map = (props: MapProps) => {
...
 
+  /**
+   * As setting onPress on the GeofencingZones ShapeSource prevents MapView's onPress
+   * from being triggered, the onPress logic is handled here instead.
+   * This makes it possible to click directly on e.g. bus stops while GeofencingZones are visible.
+   * Step 1: query all rendered features
+   * Step 2: decide feature to select
+   * Step 3: selected the feature
+   */
+
```
Comment (rosvik) [inline src/components/map/Map.tsx:135]: This extra newline annoyed me 😆

```suggestion
```

**Review by rosvik — APPROVED** [review]
